### PR TITLE
[JS] Topic: JavaScript Promises

### LINKED ISSUE #1241: JS : Promises
# Promises
---
Promises in Javascript

# Directory
---
```
Topics/10.JSON
```
 
# Description 📜
---
- What is a Promise in Javascript?
- Why do we need Promises?
- How to create a Promise? (Syntax)
- Chained Promises
- Methods in Promises
- Examples

# Domain of Contribution 📊
---
- [x] JavaScript

# Format
---
- [x] Documentation

### PR DESCRIPTION
**Pull request by 2devyank**
<hr>

## Description 📜

 This is a vedio lecture on Promises
JS:Promise in javascript -Vedio
Fixes #1241 

<hr>

## Type of change 📝

- [ ] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [x] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [ ] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev (Flutter, Java, Kotlin)
- [ ] Backend Dev (Java, .NET, Python)
- [ ] C/CPP
- [ ] Competitive Programming
- [ ] Cyber Security
- [ ] DSA
- [ ] Database
- [ ] Datascience with Python
- [ ] Datascience with R
- [ ] Frontend Dev (HTML, CSS, JS)
- [ ] Frontend Dev (React, Angular, Vue)
- [ ] Golang
- [ ] Interview Prep
- [ ] Java
- [x] JavaScript
- [ ] MERN
- [ ] Machine Learning
- [ ] Open Source
- [ ] Python
- [ ] Rust
- [ ] Statistics
- [ ] UI/UX

<hr>

## Checklist ✅

- [ ] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [ ] I have performed a self-review of my own code or work.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: JavaScript/Topics/7.Promises/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Promises In Javascript
+
+### Vedio Lecture On Promises
+Vedio-[promises](https://drive.google.com/file/d/1XRxoC3t9B_y9gARvmr95w5BfOz3-3PIf/view?usp=sharing)
```
Comment (paramsiddharth) [inline JavaScript/Topics/7.Promises/README.md:4]: Please spell `Video` correctly. The rest seems OK!

File: JavaScript/README.md [hunk 2]
```diff
@@ -17,3 +17,4 @@
   2. [Command-Line](Topics/4.Command_Line)
 - [Syntax](Topics/5.Syntax)
 - [Hoisting](Topics/6.Hoisting)
+- [Promises](Topics/7.Promises)
```
Comment (paramsiddharth) [inline JavaScript/README.md:20]: Please avoid this change, because I'll be editing the index manually and this may lead to conflicts.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

File: JavaScript/Topics/7.Promises/README.md (hunk 1 above)
Comment (2devyank) [reply JavaScript/Topics/7.Promises/README.md:4]: Oh I am really sorry @paramsiddharth

Comment (paramsiddharth) [reply JavaScript/Topics/7.Promises/README.md:4]: It's OK, Devyank. :) @2devyank You can update your PR and we will approve and merge it.

File: JavaScript/README.md (hunk 2 above)
Comment (2devyank) [reply JavaScript/README.md:20]: Actually I did not had any idea about this I am really sorry
Okay will do the changes and will make youtube link also
@paramsiddharth

**Comment by 2devyank** [comment]
@paramsiddharth I have made the necessary  changes please add label and merge it

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi @s4-2 The PR is a video-based contribution. Looks great to me! Please check out and approve so that it can be merged.

File: JavaScript/Topics/7.Promises/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Promises In Javascript
+
+### Video Lecture On Promises
+Video-[promises](https://youtu.be/ASfaBnNbeDc)
```
Comment (dakshsethi) [inline JavaScript/Topics/7.Promises/README.md:4]: It should be like this.
`Video Link: [Promises in JavaScript](https://youtu.be/ASfaBnNbeDc)`
Video Link: [Promises in JavaScript](https://youtu.be/ASfaBnNbeDc)

Also change the video title to `Promises in JavaScript`

File: JavaScript/Topics/7.Promises/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+ 
+Made with Love By-Devyank
```
Comment (dakshsethi) [inline JavaScript/Topics/7.Promises/README.md:6]: It should be like this:
`Contributor: [Devyank Nagpal](https://github.com/2devyank)`

**Review by dakshsethi — CHANGES_REQUESTED** [review]

File: JavaScript/Topics/7.Promises/README.md (hunk 4 above)
Comment (2devyank) [reply JavaScript/Topics/7.Promises/README.md:6]: Okay

**Comment by 2devyank** [comment]
@dakshsethi I have made the changes please check

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥🔥

**Review by paramsiddharth — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@Creatoon @s4-2 Please review.

**Review by Creatoon — APPROVED** [review]

**Comment by Creatoon** [comment]
@2devyank resolve conflicts

**Comment by 2devyank** [comment]
@paramsiddharth please merge this pr

**Pull request merged**